src/lib/logger.js

```diff
@@ -1,31 +1,61 @@
+const userLogger = {
+  error: undefined,
+  warn: undefined,
+  debug: undefined
+}
+
 const logger = {
-  error: (errorCode, ...text) => {
-    if (!console) { return }
+  error (code, ...text) {
     if (text && text.length <= 1) { text = text[0] || '' }
-    console.error(
-      `[next-auth][error][${errorCode.toLowerCase()}]`,
-      text,
-      `\nhttps://next-auth.js.org/errors#${errorCode.toLowerCase()}`
-    )
+    code = code.toLowerCase()
+    const link = `https://next-auth.js.org/errors#${code}`
+    if (userLogger.error) {
+      userLogger.error(code, text, link)
+    } else {
+      console.error(
+        `[next-auth][error][${code}]`,
+        text,
+        `\n${link}`
+      )
+    }
   },
-  warn: (warnCode, ...text) => {
-    if (!console) { return }
+  warn (code, ...text) {
     if (text && text.length <= 1) { text = text[0] || '' }
-    console.warn(
-      `[next-auth][warn][${warnCode.toLowerCase()}]`,
-      text,
-      `\nhttps://next-auth.js.org/warnings#${warnCode.toLowerCase()}`
-    )
+    code = code.toLowerCase()
+    const link = `https://next-auth.js.org/warnings#${code}`
+    if (userLogger.warn) {
+      userLogger.warn(code, text, link)
+    } else {
+      console.warn(
+        `[next-auth][warn][${code}]`,
+        text,
+        `\n${link}}`
+      )
+    }
   },
-  debug: (debugCode, ...text) => {
-    if (!console) { return }
+  debug (code, ...text) {
     if (text && text.length <= 1) { text = text[0] || '' }
+    code = code.toLowerCase()
+    if (userLogger.debug) {
+      userLogger.debug(code, text)
+    }
     if (process && process.env && process.env._NEXTAUTH_DEBUG) {
```

Comment:
I am not checking for `process.env._NEXTAUTH_DEBUG` here, because if the user implemented the `logger.debug` method, they probably expect it to be called. In this case, they can decide how debugging logs should be handled. (I for example have my own `process.env.LOG_LEVEL` env variable, which decides which logs should be sent to my third-pary and which one shouldn't. It also works in a way that when `process.env.NODE_ENV` is not `"production"`, instead of sending the logs, it will fall back to `console`)